src/main/java/com/zuniorteam/bowling/core/value/PitchResultType.java

```diff
@@ -0,0 +1,63 @@
+package com.zuniorteam.bowling.core.value;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import static com.zuniorteam.bowling.core.value.PinSize.ZERO;
+import static com.zuniorteam.bowling.core.value.PitchType.*;
+
+public enum PitchResultType {
+
+    GUTTER{
+        @Override
+        public boolean eq(PitchType pitchType, PinSize fallenPinSize, PinSize remainPinSize) {
+            return fallenPinSize.equals(ZERO);
+        }
+    },
+
+    MISS{
+        @Override
+        public boolean eq(PitchType pitchType, PinSize fallenPinSize, PinSize remainPinSize) {
+            return false;
+        }
+    },
+
+    SPARE{
+        @Override
+        public boolean eq(PitchType pitchType, PinSize fallenPinSize, PinSize remainPinSize) {
+            return remainPinSize.equals(ZERO) && SECOND.equals(pitchType);
+        }
+    },
+
+    STRIKE{
+        @Override
+        public boolean eq(PitchType pitchType, PinSize fallenPinSize, PinSize remainPinSize) {
+            return remainPinSize.equals(ZERO) && (FIRST.equals(pitchType) || BONUS.equals(pitchType));
+        }
+    };
+
+    public abstract boolean eq(PitchType pitchType, PinSize fallenPinSize, PinSize remainPinSize);
```

Comment:
eq에서 받는 파라미터 수를 줄여서 BiFunction을 사용하는 것이 더 좋겠지만 그게 힘들다면, 아래처럼
```java
@FunctionalInterface
public interface Eq {
    boolean eq(PitchType pitchType, PinSize fallenPinSize, PinSize remainPinSize);
}
```
펑셔널 인터페이스를 만들면
```suggestion
    private final Eq eq;

    PitchResultType(Eq eq) {
        this.eq = eq;
    }
```
```java
GUTTER((pitchType, fallenPinSize, remainPinSize) -> fallenPinSize.equals(ZERO)),
```
깔끔해 질 것 같아요